File: spring-data-day3/src/main/java/com/miu/springdataday3/entity/Product.java
package com.miu.springdataday3.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import javax.persistence.*;
import lombok.Data;

import java.util.List;

@Entity
@Data
@Table(name="products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    private String name;
    private double price;
    private double rating;

    @ManyToOne
    @JsonBackReference
    private Category category;

    @OneToMany(mappedBy = "product")
    @JsonManagedReference
    private List<Review> reviews;
}

File: lab 7/lab7/src/component/Product.js
import Category from "./Category";

const Product = (props)=>{
    return (
        <div>
            <label>Name</label>
            <span>{props.name}</span>
            <label>Price</label>
            <span>{props.price}</span>
            <label>Rating</label>
            <span>{props.rating}</span>
            {props.category && <Category category={props.category}/>}
        </div>
    )
};

export default Product;